execution_engine/src/core/engine_state/engine_config.rs

```diff
@@ -46,12 +56,14 @@ impl Default for EngineConfig {
 
 impl EngineConfig {
     /// Creates a new engine configuration with provided parameters.
+    #[allow(clippy::too_many_arguments)]
```

Comment:
This is not a god object, it's just a DTO. What we see here is legitimately configured behavior / settings being passed in rather than hardcoded or assumed.